blocks/header/header.js

```diff
@@ -1,144 +1,374 @@
+    const menuOption = {};
+    const div = navDivs[i];
+
+    // Find the h2 within this div and assign its innerHTML to the title of menuOption
+    const h2 = div.querySelector('h2');
+    if (h2) {
+      menuOption.title = h2.innerHTML;
+
+      // Find all a tags within this div
+      const links = div.querySelectorAll('a');
+      if (links.length > 0) {
+        menuOption.subMenu = [];
+
+        links.forEach((link) => {
+          const submenuItem = {};
+          // Clone the link element
+          const cloneLink = link.cloneNode(true);
+          submenuItem.name = cloneLink.textContent;
+          submenuItem.url = cloneLink.href;
+
+          submenuItem.updatedLinkHTML = cloneLink.outerHTML;
+
+          menuOption.subMenu.push(submenuItem);
+        });
+      }
+      menuOptions.push(menuOption);
+    } else {
+      // If there is no h2 in the div, remove the div from the DOM
+      div.parentNode.removeChild(div);
+    }
+  }
+
+  let originalMenuHTML;
+
+  function generateSubMenu(option) {
+    return `<div class='sub-menu-title' data-option='${option.title}'>${option.title}</div>${option.subMenu.map((subMenuItem) => `<a href='${subMenuItem.url}'>${subMenuItem.updatedLinkHTML}</a>`).join('')}`;
+  }
+
+  const menuOptionsHTML = menuOptions.map((option) => `<div class='menu-option' data-option='${option.title}'>${option.title}</div>`).join('');
+
+  // eslint-disable-next-line prefer-const
+  originalMenuHTML = menuOptionsHTML;
+
+  optionsWrapper.innerHTML = menuOptionsHTML;
+
+  function handleMenuOptionClick() {
+    const optionTitle = this.dataset.option;
+    const selectedOption = menuOptions.find((opt) => opt.title === optionTitle);
+    optionsWrapper.innerHTML = generateSubMenu(selectedOption);
+
+    const subMenuTitle = document.querySelector('.sub-menu-title');
+    if (subMenuTitle) {
+      // eslint-disable-next-line no-use-before-define
+      subMenuTitle.addEventListener('click', handleSubMenuTitleClick);
+    }
+    const optionWrapperShow = document.querySelector('.options-wrapper.show');
+    optionWrapperShow.style.height = '100vh';
+  }
+
+  function attachMenuOptionClickEvents() {
+    document.querySelectorAll('.menu-option').forEach((menuOption) => {
+      menuOption.addEventListener('click', handleMenuOptionClick);
     });
+    const optionWrapperShow = document.querySelector('.options-wrapper.show');
+    if (optionWrapperShow) {
+      optionWrapperShow.style.height = '65vh';
+    }
   }
-  // enable menu collapse on escape keypress
-  if (!expanded || isDesktop.matches) {
-    // collapse menu on escape press
-    window.addEventListener('keydown', closeOnEscape);
-  } else {
-    window.removeEventListener('keydown', closeOnEscape);
+
+  function handleSubMenuTitleClick() {
+    optionsWrapper.innerHTML = originalMenuHTML;
+    attachMenuOptionClickEvents();
   }
+
+  attachMenuOptionClickEvents();
+}
+
+async function renderMobileHeader(nav) {
```

Comment:
why is this method async?

Reply:
This was accidental, as I forgot to remove it but I have improved it